incydr/cli/cmds/alert_rules.py

```diff
@@ -0,0 +1,217 @@
+from incydr.cli.cmds.options.output_options import single_format_option
+from incydr.cli.cmds.options.output_options import SingleFormat
+from incydr.cli.cmds.options.output_options import table_format_option
+from incydr.cli.cmds.options.output_options import TableFormat
+from incydr.cli.core import IncydrCommand
+from incydr.cli.core import IncydrGroup
+from incydr.cli.render import measure_renderable
+from incydr.utils import list_as_panel
+from incydr.utils import model_as_card
+
+
+MAX_USER_DISPLAY_COUNT = 25
+
+
+@click.group(cls=IncydrGroup)
+@log_level_option
+@log_file_option
+@click.pass_context
+def alert_rules(ctx, log_level, log_file):
+    """View and manage alert rules."""
+    init_client(ctx, log_level, log_file)
+
+
+@alert_rules.command("list", cls=IncydrCommand)
+@table_format_option
+@columns_option
+@click.pass_context
+def list_(ctx: Context, format_: TableFormat = None, columns: str = None):
+    """
+    List all rules.
+    """
+    client = ctx.obj()
+    rules = client.alert_rules.v2.iter_all()
+
+    if format_ == TableFormat.table:
+        if not columns:
+            columns = [
+                "id",
+                "name",
+                "severity",
+                "is_enabled",
+                "created_at",
+                "created_by",
+                "modified_at",
+                "modified_by",
+                "description",
+            ]
+        render.table(RuleDetails, rules, columns=columns, flat=False)
+    elif format_ == TableFormat.csv:
+        render.csv(RuleDetails, rules, columns=columns, flat=True)
+    elif format_ == TableFormat.json:
+        for rule in rules:
+            console.print_json(rule.json())
+    else:
+        for rule in rules:
+            click.echo(rule.json())
+
+
+@alert_rules.command(cls=IncydrCommand)
+@click.argument("rule-id")
+@click.pass_context
+def show(ctx: Context, rule_id: str):
+    """
+    Show details for a single rule.
+
+    If using `rich`, also retrieve the username filter for the rule (if it exists).
+    """
+    client = ctx.obj()
+    rule = client.alert_rules.v2.get_rule(rule_id)
+
+    if not client.settings.use_rich:
+        click.echo(rule.json())
+        return
+
+    t = Table(title=f"Alert Rule {rule.id}")
+    t.add_column("Info")
+    tables = []
+
+    try:
+        username_filter = client.alert_rules.v2.get_users(rule_id)
+        # Pretty sure there is only ever one "user" object whose aliases indicate the usernames on the filter
+        usernames = []
+        for user in username_filter.users:
+            usernames.extend(user.aliases)
+        tables.append(list_as_panel(usernames[:MAX_USER_DISPLAY_COUNT]))
+        t.add_column(
+            f"{'Include' if username_filter.mode == '0' else 'Exclude'}d Usernames"
+        )
+    except MissingUsernameCriterionError as e:
+        raise e
+
+    if rule.notifications:
+        tables.append(model_as_card(rule.notifications))
+        t.add_column("Notifications")
```

Comment:
Splits settings objects into adjacent columns for table output, similar to what we did for `watchlists_show`